packages/astro/src/assets/utils/transformToPath.ts

```diff
@@ -33,3 +34,15 @@ export function hashTransform(
 	);
 	return shorthash(deterministicString(hashFields));
 }
+
+export function getAssetsPrefix(fileType: string, assetsPrefix?: AssetsPrefix): string {
+	if (!assetsPrefix) return '';
+	if (typeof assetsPrefix === 'string') return assetsPrefix;
+	fileType = fileType[0] === '.' ? fileType.slice(1) : fileType;
```

Comment:
nit: It looks like every usage of `getAssetsPrefix` passes `fileType` as `.js` or `.css` (for example). Perhaps we should name the param `fileExt` instead and assume that the string will always have a leading `.`? (I usually prefer doing the least work possible unless really needed.

I think we should be consistent how the "file type" is computed too, in this PR, there's different patterns:

```js
const fileExtension = extname(...)
getAssetsPrefix(fileExtension, ...)
```

```js
const fileType = extname(...)
getAssetsPrefix(fileType, ...)
```

I think we should stick with the `fileExtension` name, or `fileExt`.